background/services/enrichment/index.ts

```diff
@@ -265,80 +267,100 @@ export default class EnrichmentService extends BaseService<Events> {
 
     // Look up logs and resolve subannotations, if available.
     if ("logs" in transaction && typeof transaction.logs !== "undefined") {
-      const assets = await this.indexingService.getCachedAssets(network)
+      const subannotations = await this.parseSubAnnotations(
+        transaction.logs,
+        network,
+        desiredDecimals,
+        resolvedTime,
+        block
+      )
+
+      if (subannotations.length > 0) {
+        txAnnotation.subannotations = subannotations
+      }
+    }
 
-      const erc20TransferLogs = parseLogsForERC20Transfers(transaction.logs)
+    return txAnnotation
+  }
 
-      // Look up transfer log names, then flatten to an address -> name map.
-      const namesByAddress = Object.fromEntries(
-        (
-          await Promise.allSettled(
-            [
-              ...new Set([
-                ...erc20TransferLogs.map(
-                  ({ recipientAddress }) => recipientAddress
-                ),
-              ]),
-            ].map(
-              async (address) =>
-                [
-                  normalizeEVMAddress(address),
-                  (
-                    await this.nameService.lookUpName({ address, network })
-                  )?.name,
-                ] as const
-            )
+  async parseSubAnnotations(
+    logs: EVMLog[],
+    network: EVMNetwork,
+    desiredDecimals: number,
+    resolvedTime: number,
+    block: AnyEVMBlock | undefined
+  ): Promise<TransactionAnnotation[]> {
+    const assets = await this.indexingService.getCachedAssets(network)
+
+    const erc20TransferLogs = [
+      ...parseLogsForERC20Transfers(logs),
+      // Wrapped Deposits and Withdrawals are not actually ERC-20 transfers but we coerce them into the same type so... 🦆 Typing FTW.
+      ...parseLogsForWrappedDepositsAndWithdrawals(logs),
+    ]
+
+    // Look up transfer log names, then flatten to an address -> name map.
+    const namesByAddress = Object.fromEntries(
+      (
+        await Promise.allSettled(
+          [
+            ...new Set([
+              ...erc20TransferLogs.map(
+                ({ recipientAddress }) => recipientAddress
+              ),
+            ]),
+          ].map(
+            async (address) =>
+              [
+                normalizeEVMAddress(address),
+                (await this.nameService.lookUpName({ address, network }))?.name,
+              ] as const
           )
         )
-          .filter(
-            (result): result is PromiseFulfilledResult<[string, string]> =>
-              result.status === "fulfilled" && result.value[1] !== undefined
-          )
-          .map(({ value }) => value)
       )
+        .filter(
+          (result): result is PromiseFulfilledResult<[string, string]> =>
+            result.status === "fulfilled" && result.value[1] !== undefined
+        )
+        .map(({ value }) => value)
```

Comment:
I realize this was already the code, but a little more breaking apart might be clearer here:

```suggestion
        .filter(isFulfilledResult)
        .map(({ value }) => value)
        .filter(([, name]) => isDefined(name))
```

See https://github.com/tallycash/extension/blob/main/background/lib/utils/type-guards.ts for these two. We could even introduce a helper to extract the value from a fulfilled promise as well if we want (eg `fulfilledResultValue`).